src/fluree/db/permissions_validate.cljc

```diff
@@ -14,47 +14,50 @@
   "Returns one of:
   (a) exception if there was an error
   (b) truthy value if flake is allowed
-  (c) falsey value if flake not allowed
-
-  Note this should only be called if the db is permissioned, don't call if the root user as the results will
-  not come back correctly."
+  (c) falsey value if flake not allowed"
   [{:keys [permissions] :as db} flake]
   (go-try
-    (let [s         (flake/s flake)
-          p         (flake/p flake)
-          class-ids (or (get @(:cache permissions) s)
-                        (let [classes (<? (dbproto/-class-ids (dbproto/-rootdb db) (flake/s flake)))]
-                          ;; note, classes will return empty list if none found ()
-                          (swap! (:cache permissions) assoc s classes)
-                          classes))
-          fns       (keep #(or (get-in permissions [:view :class % p])
-                               (get-in permissions [:view :class % :default])) class-ids)]
-      (loop [[[async? f] & r] fns]
-        ;; TODO - all fns are currently sync - but that will change. Can check for presence of ch? in response, or ideally pass as meta which fns are sync or async
-        ;; return first truthy response, else false
-        (if f
-          (let [res (if async?
-                      (<? (f db flake))
-                      (f db flake))]
-            (or res
-                (recur r)))
-          false)))))
+    (if (:root? permissions)
```

Comment:
These will never be called if there is root - root is checked upstream as this code is immensely performance sensitive - so we want to avoid the call entirely.

The comment was there as a reminder, not as a TODO... so this root checking is unnecessary, and also creates confusion as it now implies it might be called with a root DB.

Reply:
> The comment was there as a reminder, not as a TODO...

I was referring to this comment: https://github.com/fluree/db/pull/352/commits/cc6b4885d05fba7e5d7a7e745443a9d92798c406#diff-2ecab4160e38dd1da4d4ac3c60747bd2fc12c186640df7519729c50e35d27e4aL48

Reply:
...but either way, if this is not desirable I'm happy to revert it.

Reply:
Not sure this file changes did anything except just add the root conditional - if not reverting would be good as it clobbers the parsing work (PR early this morning) so I'll have to redo much of that if this stays.

Reply:
https://github.com/fluree/db/pull/353